common-rest/src/main/java/org/uniprot/api/rest/download/queue/RabbitMQConfig.java

```diff
@@ -0,0 +1,116 @@
+package org.uniprot.api.rest.download.queue;
+
+import org.springframework.amqp.core.AmqpAdmin;
+import org.springframework.amqp.core.Binding;
+import org.springframework.amqp.core.BindingBuilder;
+import org.springframework.amqp.core.DirectExchange;
+import org.springframework.amqp.core.Exchange;
+import org.springframework.amqp.core.ExchangeBuilder;
+import org.springframework.amqp.core.Queue;
+import org.springframework.amqp.core.QueueBuilder;
+import org.springframework.amqp.rabbit.connection.CachingConnectionFactory;
+import org.springframework.amqp.rabbit.connection.ConnectionFactory;
+import org.springframework.amqp.rabbit.core.RabbitAdmin;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+
+/**
+ * Initialisation code for Rabbit MQ to be used by both producer and consumer
+ *
+ * @author sahmad
+ * @created 23/11/2022
+ */
+@Configuration
+@Profile({"asyncDownload"})
+public class RabbitMQConfig {
+
+    @Bean
+    public RabbitMQConfigProperties rabbitMQConfigProperties() {
+        return new RabbitMQConfigProperties();
+    }
+
+    @Bean
+    public ConnectionFactory connectionFactory(RabbitMQConfigProperties rabbitMQConfigProperties) {
+        CachingConnectionFactory connectionFactory =
+                new CachingConnectionFactory(rabbitMQConfigProperties.getHost());
+        connectionFactory.setUsername(rabbitMQConfigProperties.getUser());
+        connectionFactory.setPassword(rabbitMQConfigProperties.getPassword());
+        connectionFactory.setPort(rabbitMQConfigProperties.getPort());
+        return connectionFactory;
+    }
+
+    @Bean
+    public Exchange downloadExchange(RabbitMQConfigProperties rabbitMQConfigProperties) {
```

Comment:
Isn't this download agnostic? In that case, you can stick to a more generic name s.t. directExchange

Reply:
thanks